npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/registry_finder.rb

```diff
@@ -86,6 +86,10 @@ def first_registry_with_dependency_details
                 headers: auth_header_for(details["token"])
               )
               response.status < 400 && JSON.parse(response.body)
+              if response.status >= 400
+                raise DependencyFileNotResolvable,
```

Comment:
can you dig into where `known_registries` is set? It should only be a private registry if a private registry is set